### ISSUE TITLE
nwjs 0.18.0 MacOS 10.11.6 frozen after some page reload

### ISSUE THREAD

**Issue by AndryBray** [nwjs/nw.js#5386]
I have again problems with reloading app (cmd+r) on nwjs with sdk.

RAM usage is growing up  after every reloading and the window viewport will be locked and unresponsive. 

You have to try reload a complex app (not just a simple website), something like gmail/youtube or some big crm.

**Comment by AndryBray** [comment]
It happens even without reloading. Just using it intensively (last 10 tens crashed with 600/900Mb of RAM usage)

**Comment by oaleynik** [comment]
@AndryBray I've seen this as well. For me it happens when I actively use DevTools

**Comment by AndryBray** [comment]
Crashed again. This is what I see inside the mac console log is:

```
20/10/16 12:37:21,000 kernel[0]: nwjs Helper[1086] triggered unnest of range 0x7fff91200000->0x7fff91400000 of DYLD shared region in VM map 0xa9ec2d0436184bd9. While not abnormal for debuggers, this increases system memory footprint until the target exits.
20/10/16 12:39:46,000 kernel[0]: nwjs Helper[1087] triggered unnest of range 0x7fff91200000->0x7fff91400000 of DYLD shared region in VM map 0xa9ec2d04334da9e9. While not abnormal for debuggers, this increases system memory footprint until the target exits.
20/10/16 12:39:47,000 kernel[0]: nwjs Helper[1088] triggered unnest of range 0x7fff91200000->0x7fff91400000 of DYLD shared region in VM map 0xa9ec2d043985f229. While not abnormal for debuggers, this increases system memory footprint until the target exits.
```

**Comment by AndryBray** [comment]
@oaleynik Yes, I use devtools a lot, I log almost every method called inside my app for debug reason. Didn't test yet too much without devtools. Do you think is something related to `console.log` method?

**Comment by oaleynik** [comment]
@AndryBray possible. Or something about DevTools at all. I didn't see the freezing when DevTools are not opened though. 

Another strange thing I noticed - sometimes when DevTools are opened and you hit Reload - DevTools will show multiple copies of the same files.

**Comment by AndryBray** [comment]
@oaleynik @rogerwang  It has to be something wrong with devtools and garbage collector, memory shouldn't increase after reloads.

**Comment by AndryBray** [comment]
ram usage increases even on windows (tested right now on Window 7 / 64bit)

**Comment by renemarty** [comment]
Any news on this? It's a serious bug of nw... I'm analysing to migrate a huge app to electron only due to this memory issue. Any advice or comment will be appreciated.

**Comment by AndryBray** [comment]
@renemarty did you test it with latest 0.19.x ?

**Comment by renemarty** [comment]
Tested last week with nw.js 0.19.4, Node 7.3.0, Chromium 55.0.2883.87 en Mac OS X

**Comment by rogerwang** [comment]
Please make sure there is no reference to the DOM objects in the Node context. The Node context lives in the background page and keeps running throughout the application's lifetime. To do that use the heap profiler in the devtools to find out what objects is referred.

Another way is to use the "mixed context" mode, where Node context and DOM context is the same.

See more here: http://docs.nwjs.io/en/latest/For%20Users/Advanced/JavaScript%20Contexts%20in%20NW.js/
